src/core/ipns/resolver.js

```diff
@@ -89,19 +89,33 @@ class IpnsResolver {
 
   // resolve ipns entries from the provided routing
   _resolveName (name, callback) {
-    const peerId = PeerId.createFromB58String(name)
+    let peerId
+
+    try {
+      peerId = PeerId.createFromB58String(name)
+    } catch (err) {
+      return callback(err)
+    }
+
     const { routingKey } = ipns.getIdKeys(peerId.toBytes())
 
     // TODO DHT - get public key from routing?
     // https://github.com/ipfs/go-ipfs/blob/master/namesys/routing.go#L70
     // https://github.com/libp2p/go-libp2p-routing/blob/master/routing.go#L99
 
     this._routing.get(routingKey.toBuffer(), (err, res) => {
-      if (err || !res) {
+      if (err) {
         const errMsg = `record requested was not found for ${name} (${routingKey}) in the network`
 
         log.error(errMsg)
-        return callback(errcode(new Error(errMsg), 'ERR_NO_NETWORK_RECORD_FOUND'))
+        return callback(errcode(new Error(errMsg), 'ERR_NO_RECORD_FOUND'))
+      }
+
+      if (!res) {
+        const errMsg = `record requested was empty for ${name} (${routingKey}) in the network`
+
+        log.error(errMsg)
+        return callback(errcode(new Error(errMsg), 'ERR_EMPTY_RECORD_FOUND'))
       }
```

Comment:
I was thinking a `null` value for `res` would signify not found...but it looks like that's not the case?

If an error is raised for not found then I don't think we need to deal with this - I'd remove this and let it continue through to `Record.deserialize`/`ipns.unmarshal` where it will fail...

Reply:
agree that differentiating the errors above is enough